frontend/controller/actions/group.js

```diff
@@ -897,6 +897,61 @@ export default (sbp('sbp/selectors/register', {
       sbp('okTurtles.events/emit', OPEN_MODAL, 'AddMembers')
     }
   },
+  'gi.actions/group/removeUselessIdentityContracts': function ({ contractID, possiblyUselessContractIDs }) {
+    // NOTE: should remove the identity contracts which we don't need to sync anymore
+    //       for users who don't have any common groups, and any common DMs
+    const interval = setInterval(() => {
```

Comment:
I'm not a fan of using `setInterval` in general, and in this case it seems like it might work but is unsafe. If you must do it this way, firstly I'd use setTimeout instead, which ensures that the callback isn't executed more than once at the same time. I'd also combine it with a `/wait` call to start it off, since you're expecting that those contracts to be removed, so most of the time there won't be any more events after the call to `/wait` completes.

A safer approach to doing this, without any `setTimeout` / `setInterval` would be to create a new sync barrier API. This barrier API would be something of a _semaphore_ that'd also keep track in its internal state what's being waited, to attempt to break cycles in some way.